chromium_src/components/permissions/permission_request.cc

```diff
@@ -19,14 +21,20 @@ bool PermissionRequest::SupportsLifetime() const {
   return false;
 }
 
-void PermissionRequest::SetLifetime(base::Optional<base::TimeDelta> lifetime) {
+void PermissionRequest::SetLifetime(absl::optional<base::TimeDelta> lifetime) {
   DCHECK(SupportsLifetime());
   lifetime_ = std::move(lifetime);
 }
 
-const base::Optional<base::TimeDelta>& PermissionRequest::GetLifetime() const {
+const absl::optional<base::TimeDelta>& PermissionRequest::GetLifetime() const {
   DCHECK(SupportsLifetime());
   return lifetime_;
 }
 
+// Needs to be synced with upstream's PermissionRequest::IsDuplicateOf().
+bool PermissionRequest::IsDuplicateOf(PermissionRequest* other_request) const {
+  return GetRequestType() == other_request->GetRequestType() &&
```

Comment:
maybe we can call ```PermissionRequest_ChromiumImpl::IsDuplicateOf_Unused``` here?

Reply:
I tried that and it works. It's better to use define:
```#define IsDuplicateOf IsDuplicateOf_ChromiumImpl```
and then do this:
```
bool PermissionRequest::IsDuplicateOf(PermissionRequest* other_request) const {
  return PermissionRequest_ChromiumImpl::IsDuplicateOf_ChromiumImpl(
      other_request);
}
```

in header modify comment to say: "We **rename** upstream's IsDuplicateOf()..."
and remove comment in .cc file at all.

Reply:
Agree on reusing the upstream implementation. Should lifetime_ member of our subclass figure into `IsDuplicateOf` - that is, would 2 requests with different lifetimes constitute the same request?

Reply:
> Should lifetime_ member of our subclass figure into IsDuplicateOf - that is, would 2 requests with different lifetimes constitute the same request?

no, I don't think so. lifetimes is not a part of a request (it doesn't come from a requesting side).

Reply:
Addressed in 45758c6